spec/features/finance/npq/course_payment_breakdown_spec.rb

```diff
@@ -19,28 +19,23 @@
     )
   end
 
-  scenario "Can get to NPQ payment breakdown page for a provider", :js do
+  scenario "can get to NPQ payment breakdown page for a provider" do
     given_i_am_logged_in_as_a_finance_user
     and_there_is_npq_provider_with_contracts
     and_those_courses_have_submitted_declations
     when_i_visit_the_payment_breakdown_page
     and_choose_to_see_npq_payment_breakdown
     and_i_select_an_npq_lead_provider
     then_i_should_have_the_correct_payment_breakdown_per_npq_lead_provider
-    when_i_click_on(npq_leading_teaching_contract)
 
-    then_i_should_see_correct_breakdown_summary(cpd_lead_provider, npq_leading_teaching_contract)
-    then_i_should_see_correct_payment_breakdown(npq_leading_teaching_contract)
-    when_i_click "Back"
-
-    when_i_click_on(npq_leading_behaviour_culture_contract)
-    then_i_should_see_correct_breakdown_summary(cpd_lead_provider, npq_leading_behaviour_culture_contract)
-    then_i_should_see_correct_payment_breakdown(npq_leading_behaviour_culture_contract)
-    when_i_click "Back"
-
-    when_i_click_on(npq_leading_teaching_development_contract)
-    then_i_should_see_correct_breakdown_summary(cpd_lead_provider, npq_leading_teaching_development_contract)
-    then_i_should_see_correct_payment_breakdown(npq_leading_teaching_development_contract)
+    [npq_leading_teaching_contract, npq_leading_behaviour_culture_contract, npq_leading_teaching_development_contract].each do |npq_contract|
+      when_i_click_on(npq_contract)
```

Comment:
I would probably create a separate scenario for 31 -37 since this scenario only covers getting to the page which is the first paragraph, not essential though

Reply:
I'm usually reluctant to separate checking each pages in a single scenario, launching a test takes a fare bit of time and slows down the test suite a substantial amount.
In this case getting to a course payment breakdown page involves getting through the courses page in the first place, I could just `visit course_payment_breakdown_path(...params)`. But these sort of belong to same _flow_.
Maybe one issue I could this approach it hides a bit the testing of single course payment, I'm happy to break it down. Thoughts? 
I update the scenario description do clearly describe its content.

Reply:
Yeah I totally get what you are saying, think updating the scenario description is fine here 👍